OCPBUGS-49725#Add info on how to define subnet/ip access restrictions… (#90251)
openshift/openshift-docs: modules/persistent-storage-csi-manila-dynamic-provisioning.adoc

. Define the required options on the resulting page.

.. Select the appropriate storage class.
+
[IMPORTANT]
====

> To limit the clients that can mount the persistent volume (PV), set an IP or a subnet mask in the `nfs-shareClient` storage class parameter.

Reviewer: @gouthampacha @silvacarloss 
Does manila operator reconcile the SCs it manages if the admin makes a change to them?
Do we need to add a link on how to disable reconciliation?
https://docs.redhat.com/en/documentation/openshift_container_platform/4.16/html/storage/using-container-storage-interface-csi#persistent-storage-csi-sc-manage

Reviewer 2: Good catch, I believe it is not the pre-defined storageclass created by operator, user should create a own sc with the parameter. I think it's better to mention it explicitly.

I think maybe we can put it in the end of the section and mention something like:

> To limit the clients that can mount the persistent volume (PV), please create a new storageclass with setting an IP or a subnet mask in the `nfs-shareClient` storage class parameter.

Author: Fixed! @gcharot please provide LGTM, if everything looks OK.

Reviewer: @duanwei33 Do we want to provide the two alternatives?
Create a new one or edit the pre-defined[1]

[1] https://docs.redhat.com/en/documentation/openshift_container_platform/latest/html/storage/using-container-storage-interface-csi#persistent-storage-csi-sc-manage

Reviewer 2: @gcharot 
It is not allowed to update the storageclass.parameters: 
> storageclasses.storage.k8s.io "csi-manila-ceph" was not valid:
> parameters: Forbidden: updates to parameters are forbidden.

The Manila storage class is not set as the default (as per the default OCP configuration), so users always need to specify the storage class name in the PVC. Therefore, adding a new storage class should be sufficient. Of course, users can configure a unique Manila storage class using the nfs-shareClient parameter if needed (and removing pre-defined sc), but that decision could be ultimately up to the customer.

Reviewer 2: @lpettyjo We don't mention new sc creation here, actually it might be late while we are creating a PVC.
My suggestion is we only add this as a generic tip to mention the parameter(at the beginning or end of the section), not inside the steps - The same as the CLI part.

Reviewer: yes agree. 
Add a note at the beginning mentioning that by default the access rule is 0.0.0.0/0
You need to create a new SC and use the nfs-shareClient to restrict the IP/Subnets.

Reviewer 3: Apologies for the late response.
> Does manila operator reconcile the SCs it manages if the admin makes a change to them?

Yes, it does. Goutham and I tested it and it currently reconciles the SCs.

> Add a note at the beginning mentioning that by default the access rule is 0.0.0.0/0
> You need to create a new SC and use the nfs-shareClient to restrict the IP/Subnets.

++

> To limit the clients that are able to mount the persistent volume (PV), set an IP or a subnet mask in the `nfs-shareClient` storage class parameter.

Reviewer 4: Nit:
```suggestion
To limit the clients that can mount the persistent volume (PV), set an IP or a subnet mask in the `nfs-shareClient` storage class parameter.
```

  storageClassName: csi-manila-gold <2>
----
+
<1> Use RWX if you want the PV that fulfills this PVC to be mounted to multiple pods on multiple nodes in the cluster.
<2> The name of the storage class that provisions the storage back end. Manila storage classes are provisioned by the Operator and have the `csi-manila-` prefix.
+
[IMPORTANT]
====

> To limit the clients that are able to mount the PV, set an IP or a subnet mask in the `nfs-shareClient` storage class parameter.

Reviewer 4: Nit:
```suggestion
To limit the clients that canmount the PV, set an IP or a subnet mask in the `nfs-shareClient` storage class parameter.
```
